### DISCUSSION TITLE
Can XHTML be used with Leptos?

### DISCUSSION THREAD

**Discussion by kellytk** [leptos-rs/leptos#689] [Q&A]
I've worked extensively with Yew however I'm now evaluating alternatives.  I prefer the XHTML syntax of HTML(5).  Can I understand the mention of XHTML at https://github.com/leptos-rs/leptos/blob/55ce805b60ce27ca4f475206c00665e7da2923c2/leptos_macro/src/lib.rs#L53 as explicit support of XHTML by Leptos?

**Comment by gbj** [comment]
I'm not entirely sure I understand what you mean when you say "I prefer the XHTML syntax of HTML(5)." Do you mean you prefer XHTML _over_ HTML5? The Leptos `view` macro does require things like self-closing tags, as in XHTML rather than HTML, but it is not identical with XHTML. I'd recommend taking a look at some of the examples to understand what the syntax looks like.

**Reply by kellytk** [reply]
Authoring in [RSX](https://github.com/leptos-rs/leptos/blob/55ce805b60ce27ca4f475206c00665e7da2923c2/leptos_macro/src/lib.rs#L40) is acceptable as long as whatever is provided to the browser is strictly conforming XHTML.

**Reply by gbj** [reply]
Sorry, you're talking about what is generated by server-side HTML rendering, not the view macro you're using in Rust? The text you posted is about the syntax of the view macro.

Frontend frameworks of course don't generate either HTML or XHTML but create DOM nodes directly. Leptos and Yew also both support server-side rendering of HTML. I can't guarantee that what Leptos generates is "strictly conforming XHTML" as opposed to HTML5. Why would it make a difference to you, just curious?

**Reply by kellytk** [reply]
I was thinking in the context of client-side generation, however I had forgotten that frontend frameworks directly manipulate the DOM. That's probably then a non-issue and thank you for the reminder.

Server-side generation is more likely where I'll need to verify strict XHTML conformance, so as to avoid browsers displaying a syntax error rather than the intended app.

I'll experiment and update this discussion with my findings.

Thank you for your prompt and helpful responses!

**Reply by gbj** [reply]
You're welcome, and sounds good! Modern browsers do not have an issue with well-formed HTML, whether it's XHTML or not. If you find a situation in which the browser displays a syntax error, *definitely* open an issue and we'll fix it ASAP.

Enjoy!
